api/services/freshtrakService.ts

```diff
@@ -0,0 +1,53 @@
+import { FRESHTRAK_ZIP_URL, FRESHTRAK_AGENCY_URL } from '../utils/constants';
+import freshtrakEventDto from '../models/freshtrakAPI/freshtrakEventDto';
+import freshtrakLocationDto from '../models/freshtrakAPI/freshtrakLocationDto';
+import { getFreshTrakEvents } from './freshtrakApiService';
+
+async function getFTLocationData(
+  siteId: string,
+  zip: string
+): Promise<freshtrakLocationDto> {
+  // in memory map to link siteId to FreshTrak agency id
+  // ToDo add linkage in FreshTrak Agencies endpoint
+  const freshTrakAgencies = new Map();
```

Comment:
:sunglasses: just fyi I moved this here to avoid a memory leak from a variable being outside of the function's scope